src/parser.ts

```diff
@@ -29,42 +37,42 @@ export const ZodParser = (schemaDef: z.ZodTypeDef) => (obj: any, params: ParsePa
 
   switch (def.t) {
     case z.ZodTypes.string:
-      if (typeof obj !== 'string') throw ZodError.fromString(`Non-string type: ${typeof obj}`);
+      if (typeof obj !== 'string') ZodErrorHandler(def.customError || `Non-string type: ${typeof obj}`);
```

Comment:
maybe pass `customError` as closure?

```
const zodErrorHandler = createZodErrorHandler(def.customError);
```

Reply:
I will overthink it do you mean it will allow to clean stacktrace ?